internal/integration-corpus-generator/client.go

```diff
@@ -0,0 +1,75 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package integration_corpus_generator
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/logger"
+)
+
+const integrationCorpusGeneratorAssetsBaseURL = "https://raw.githubusercontent.com/elastic/elastic-integration-corpus-generator-tool/main/assets/templates/"
```

Comment:
this is a temporary solution in place before we will have the relevant assets content from the integration corpus generator in package-spec.

until then I'm unsure how and if we should manage the version/tag of the assets: any extra complexity related to this will be finally removed once we can rely on package-spec, so I'd rather assume `main` branch and do not handle version/tag of the assets for the moment

Reply:
Please add a `TODO`/`FIXME` comment here about this.

Reply:
Could it be set to a specific tag or reference instead of `main`? That would help to identify which templates are used in each elastic-package version.

Reply:
> Could it be set to a specific tag or reference instead of `main`? That would help to identify which templates are used in each elastic-package version.

I completely understand and agree with the value of identify which template are used in each elastic-package version, but the only way to reference a specific tag or reference instead of main is either:
1. pass the tag as argument of the command, so that's resolved at runtime
2. keep the tag in the elastic package codebase: this will require that whenever we add new templates on the corpus generator repo we need to file a tag, updated in the elastic package code and release a new elastic package version

I pinned main exactly to avoid 2 and at the same time being able to "consume" the artifacts as soon as they are merged

since fetching the artifacts from the corpus generator repo it's just a temporary solution I would also not introduce 1. since it will be later removed: when the information contained by the artifacts will be part of package-specs we'll have to define the integrations version, rather than the tag in the corpus generator repo.

what do you think about going for 1., using `package-version` or similar, so that we can later use the same flag to just execute a new logic?

it will be a little misleading while we are in the intermediate phase, but future proof

Reply:
@aspacca Thanks for sharing these options and information!

As you mentioned the idea is to have this as part of the package-spec repo, and if we add that new parameter to specify the tag, then it should be removed afterwards (once it is in the package-spec)... I think it can be kept as using `main` as it does not affect currentcly CI, but as @jsoriano suggested adding that TODO or FIXME comment.